internal/services/devcenter/dev_center_catalog_data_source.go

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MPL-2.0
+
+package devcenter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2025-02-01/catalogs"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2025-02-01/devcenters"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.DataSource = DevCenterCatalogDataSource{}
+
+type DevCenterCatalogDataSource struct{}
+
+type DevCenterCatalogDataSourceModel struct {
+	Name              string                             `tfschema:"name"`
+	ResourceGroupName string                             `tfschema:"resource_group_name"`
+	DevCenterID       string                             `tfschema:"dev_center_id"`
+	CatalogGitHub     []CatalogPropertiesDataSourceModel `tfschema:"catalog_github"`
+	CatalogAdoGit     []CatalogPropertiesDataSourceModel `tfschema:"catalog_adogit"`
+}
+
+type CatalogPropertiesDataSourceModel struct {
+	URI            string `tfschema:"uri"`
+	Branch         string `tfschema:"branch"`
+	KeyVaultKeyUrl string `tfschema:"key_vault_key_url"`
+	Path           string `tfschema:"path"`
+}
+
+func (DevCenterCatalogDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+
+		"dev_center_id": commonschema.ResourceIDReferenceRequired(&devcenters.DevCenterId{}),
+	}
+}
+
+func (DevCenterCatalogDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"catalog_github": CatalogPropertiesSchemaForDataSource(),
+
+		"catalog_adogit": CatalogPropertiesSchemaForDataSource(),
+	}
+}
+
+func (DevCenterCatalogDataSource) ModelObject() interface{} {
+	return &DevCenterCatalogDataSourceModel{}
+}
+
+func (DevCenterCatalogDataSource) ResourceType() string {
+	return "azurerm_dev_center_catalog"
+}
+
+func (r DevCenterCatalogDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DevCenter.V20250201.Catalogs
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var state DevCenterCatalogDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			devCenterId, err := catalogs.ParseDevCenterID(state.DevCenterID)
+			if err != nil {
+				return fmt.Errorf("parsing dev center id: %+v", err)
```

Comment:
The `ParseDevCenterID` function already returns an error message in the form of `parsing <input>: <error>` so we don't need to wrap it here

```suggestion
				return err
```

Reply:
Updated

Reply:
Updated